testsupport/init.go

```diff
@@ -105,6 +108,74 @@ func waitForOperators(t *testing.T) {
 	t.Log("all operators are ready and in running state")
 }
 
+func getE2EServiceAccountToken(t *testing.T, hostNs string, apiConfigsa *api.Config, sacl client.Client) string {
+
+	// creating another config which is used for creating only resclient,
+	//so that the main kubeconfig is not altered
+	restkubeconfig, err := util.BuildKubernetesRESTConfig(*apiConfigsa)
+	require.NoError(t, err)
+
+	// Check if there is already a service account present for e2e test
+	sa := &corev1.ServiceAccount{}
+	err = sacl.Get(context.TODO(), types.NamespacedName{Namespace: hostNs, Name: "e2e-test"}, sa)
+
+	sacrb := &rbacv1.ClusterRoleBinding{}
+	err1 := sacl.Get(context.TODO(), types.NamespacedName{Namespace: hostNs, Name: "e2e-test"}, sacrb)
+	// If not found proceed to create the e2e service account and the cluster role binding
+	if errors.IsNotFound(err) {
+		t.Logf("No Service Account for e2e test found, proceeding to create it")
+		err := sacl.Create(context.TODO(), &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "e2e-test",
+				Namespace: hostNs}})
+		require.NoError(t, err, "Error in creating Service account for e2e test")
+
+		crb := rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "e2e-test-cluster-admin",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     "cluster-admin",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      "e2e-test",
+					Namespace: hostNs,
+				},
+			},
+		}
+		// check if there are any clusterrolebinding present from the previous run of e2e test
+		if errors.IsNotFound(err1) {
+			t.Logf("Proceeding to create Cluster Role Binding for the Service Account")
+			err = sacl.Create(context.TODO(), &crb)
+			require.NoError(t, err, "Error in Creating Cluster role binding")
+		}
+		t.Logf("There is already a ClusterroleBinding for e2e-test")
+
+	} else if err != nil {
+		require.NoError(t, err, "Error fetching service accounts")
+	}
```

Comment:
ok, let's not over-complicate it and do the steps one-by-one - at first ensure SA and then ClusterRoleBinding (no nesting needed, nor mix of instances of the errors)
```go
	sa := &corev1.ServiceAccount{}
	err = sacl.Get(context.TODO(), types.NamespacedName{Namespace: hostNs, Name: "e2e-test"}, sa)
	// If not found proceed to create the e2e service account and the cluster role binding
	if errors.IsNotFound(err) {
		t.Logf("No Service Account for e2e test found, proceeding to create it")
		err := sacl.Create(context.TODO(), &corev1.ServiceAccount{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "e2e-test",
				Namespace: hostNs}})
		require.NoError(t, err, "Error in creating Service account for e2e test")
	} else if err != nil {
		require.NoError(t, err, "Error fetching service accounts")
	}

	sacrb := &rbacv1.ClusterRoleBinding{}
	err = sacl.Get(context.TODO(), types.NamespacedName{Namespace: hostNs, Name: "e2e-test-cluster-admin"}, sacrb)
	// check if there are any clusterrolebinding present from the previous run of e2e test
	if errors.IsNotFound(err) {
		crb := rbacv1.ClusterRoleBinding{
			ObjectMeta: metav1.ObjectMeta{
				Name: "e2e-test-cluster-admin",
			},
			RoleRef: rbacv1.RoleRef{
				APIGroup: "rbac.authorization.k8s.io",
				Kind:     "ClusterRole",
				Name:     "cluster-admin",
			},
			Subjects: []rbacv1.Subject{
				{
					Kind:      "ServiceAccount",
					Name:      "e2e-test",
					Namespace: hostNs,
				},
			},
		}
		t.Logf("Proceeding to create Cluster Role Binding for the Service Account")
		err = sacl.Create(context.TODO(), &crb)
		require.NoError(t, err, "Error in Creating Cluster role binding")
	} else if err != nil {
		require.NoError(t, err, "Error fetching clusterrolebinding")
	}
```

Reply:
Missed that.. thanks for pointing it out. changed it now